hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -214,6 +218,20 @@ private static GenericRecord rewrite(GenericRecord record, Schema schemaWithFiel
     return newRecord;
   }
 
+  /*
+  This function takes the union of all the fields except hoodie metadata fields
```

Comment:
nit: formatting , missing `*`

Reply:
let's avoid overloading the term "union" here.. "Generates a super set of fields from both old and new schema"